test/release/examples/primers/modules.chpl

```diff
@@ -0,0 +1,390 @@
+       private variable hiddenFoo and the private function hiddenBaz.  However,
+       any private symbol defined within Inner2 will not be visible to symbols
+       defined outside of Inner2.
+     */
+
+    private var innerOnly = -17;
+    var canSeeHidden = !hiddenFoo;
+  }
+
+  // At the moment, private cannot be applied to type definitions; type
+  // aliases, and declarations of enums, records, and classes cannot be declared
+  // private.  Private also cannot be applied to fields or methods yet.
+
+}
+
+module AnotherModule {
+  var a = false;
+}
+
+module ThirdModule {
+  var b = -13.0;
+}
+
+
+module Conflict {
+  /* This parenthesis-less function shares a name with a symbol in modToUse. */
+  proc bar {
+    writeln("In Conflict.bar");
+    return 5;
+  }
+
+  var other = 5.0 + 3i;
+}
+
+
+proc main() {
+  writeln("Access from outside a module");
+
+  /* Often, it is desirable for the contents of a module to be accessed from
+     outside that context.  There are several strategies for accomplishing this:
+
+     First, if only one symbol defined in the module is desired, and it is
+     desired only once, the symbol can be referenced explicitly - this is done
+     using the module name and a separating '.' as a prefix to the name of the
+     symbol desired.
+  */
+  var thriceFoo = 3 * modToUse.foo; // should be 36
+  writeln(thriceFoo);
+
+
+  {
+    /* If many of the module's symbols are desired, or the same symbol is
+       desired multiple times, then it is best to utilize what is known as a
+       "use statement".
+    */
+    use modToUse;
+
+    /*
+      Use statements can be inserted at the global scope, as well as any
+      arbitrary inner scope defined.
+    */
+
+    /* A use statement brings all of the visible symbols of a module into the
+       scope which defines the use statement.  These symbols may then be
+       accessed without the module name prefix.
+    */
+    var bazBarFoo = baz(bar, foo); // should be 28
+    writeln(bazBarFoo);
+  }
+
+
+
+  /* Once the scope defining a use statement is left, however, the symbols must
+     be accessed with a prefix.  The following line would cause an error,
+     because the previous use of the module 'modToUse' occurred within the curly
+     braces above:
+  */
+  // var twiceFoo = 2 * foo;
+
+
+
+  {
+    var bazBarFoo = baz(bar, foo); // should be 28
+
+    /* Use statements apply to the entire scope in which they are defined.  Even
+       if the use statement occurs after code which would access the symbols
+       without prefix, these accesses are still valid.
+    */
+    use modToUse;
+
+    writeln(bazBarFoo);
+  }
+
+
+  {
+    /* The symbols provided by a use statement are considered at an outer scope
+       to the symbols defined at the same level as the use itself.  Thus, if
+       another bar were defined here, and an access was attempted, the compiler
+       would find the bar at this scope, rather than modToUse.bar.
+    */
```

Comment:
"are considered ... as the use itself." -- for simplicity, what about replacing this with something like "are only considered when symbols that cannot be resolved directly within the local scope."  While I know I often talk about things like injecting a scope outside of the current scope, I think it's confusing to try and describe that way.

Isn't it also the case that statements like this one only apply to symbols that shadow others, like variables and constants, but not to things like functions where overloads at all visible scopes will be considered?  Is there some way we can reflect that here without making things overly complicated?